cmd_watch.go

```diff
@@ -126,158 +124,48 @@ func waitCircle(parent context.Context, cfg watchConfig) (bool, time.Time, error
-			if finished {
-				checksLeft--
-				if checksLeft <= 0 {
-					// we're done checking.
-					passed = !failed
-					if passed {
-						fmt.Println("Build passed!")
-					} else {
-						fmt.Println("Build failed!")
-					}
-					return
-				}
-				if err != nil {
-					// we previously successfully queried for the workflow; this is likely a
-					// transient error
-					fmt.Printf("Querying the CirlceCI API failed with %s; trying %d more times before giving up.\n", err.Error(), checksLeft)
-					continue
-				}
-				if failed {
-					// don't bother rechecking if the job has failed but didn't error
-					fmt.Printf("Build failed!\n")
-					return
-				}
-				// yay looks like maybe we're done?
-				fmt.Printf("Build appears finished; checking %d more times to make sure.\n", checksLeft)
+		wf, err = client.GetWorkflowV2(cfg.workflowID)
+		if err != nil {
+			// we previously successfully queried for the workflow; this is likely a transient error
+			fmt.Printf("\nQuerying the CirlceCI API failed with %s;", err.Error())
+			checksLeft--
+			if checksLeft > 0 {
+				fmt.Printf(" trying %d more times before giving up.\n", checksLeft)
 				continue
 			}
-			// if we previously thought we were finished but now realize we weren't,
-			// reset the check counter so we try 3 times again next time we think we're
-			// finished.
-			passed = false
-			checksLeft = numChecks
-
+			fmt.Println()
+			return false, start, nil
 		}
-	}()
+		checksLeft = numChecks
+		status := wf.Status
 
-	<-done
-	return passed, start, nil
-}
-
-// evalWorkflow looks at the CircleCI API for the list of jobs in this workflow
-// and decides whether the build has finished and if finished, whether it
-// failed. If an error is returned, it represents an error talking to the
-// CircleCI API, not an error with the workflow.
-func evalWorkflow(client *circleci.Client, wfID string, jobName string) (finished bool, failed bool, err error) {
-	fmt.Printf("%s: polling for jobs: ", time.Now().Format(time.StampMilli))
-	wfJobs, err := getJobs(client, wfID)
-	if err != nil {
-		fmt.Printf("error polling: %s\n", err.Error())
-		return true, true, err
-	}
-	fmt.Println(summarizeJobList(wfJobs))
-
-	for _, job := range wfJobs {
-		// always count ourself as finished so we don't wait for ourself
-		if job.Name == jobName {
-			continue
-		}
-
-		switch job.Status {
-		case "success", "blocked":
-			// success means it passed
-			// blocked means it can't yet run, but that could be because either
-			// it's waiting on a running job or
-			// it's not configured to run this build (because of a tag or something)
-			continue
-		case "queued":
-			return false, failed, nil
-		case "failed":
-			failed = true
+		switch status {
+		case "running", "failing":
+			fmt.Print(".")
 			continue
-		case "running":
-			// We can stop short as soon as we find an unfinished job
-			return false, failed, nil
+		case "success":
+			fmt.Println("\nBuild passed!")
+			return true, start, nil
+		default:
```

Comment:
I think you'll want to handle the `on_hold` status differently too - that happens when there's an unapproved approval job.